document_merge_service/api/engines.py

```diff
@@ -163,6 +163,12 @@ def validate_template_syntax(self, available_placeholders=None, sample_data=None
             arg_str = ";".join(exc.args)
             raise exceptions.ValidationError(f"Syntax error in template: {arg_str}")
 
+        except TypeError as exc:
+            if "unsupported operand type(s) for -" in exc.args[0]:
+                raise exceptions.ValidationError(
+                    'Syntax error in template: "-" is not allowed in placeholders'
+                )
+
```

Comment:
You could probably generalize this a bit more - This looks like a rather specific error, and most importantly, if it's some other form of `TypeError`, the code just continues on.

At the minimum, add the following lines:

```suggestion
        except TypeError as exc:
            if "unsupported operand type(s) for -" in exc.args[0]:
                raise exceptions.ValidationError(
                    'Syntax error in template: "-" is not allowed in placeholders'
                )
            else:
                raise
```

Reply:
For example, what about other characters / operators like `/`, `+` and so on?